scripts/lib_refactor/README.md

```diff
@@ -0,0 +1,36 @@
+# hal_gigadevice lib refactor
```

Comment:
This docs must be linked with [Main README](https://github.com/zephyrproject-rtos/hal_gigadevice/blob/63a72ca90b7e0d7257211ddc5c79e8c0b940371b/README.md) because there are mentions there about procedures to submit code. Everything that is handled by `cocci` could be dropped from Main Readme and stay here.